src/main.rs

```diff
@@ -247,7 +260,28 @@ impl App {
     fn next(&self, args: &ArgMatches) -> Result<()> {
         let next_handle = self
             .handle_from_args(args)
-            .or_else(|_| self.random_handle())?;
+            .or_else(|_| {
+                if args.get_flag("reverse") {
+                    let max_attemps = 100;
+                    for _i in 0..max_attemps {
```

Comment:
I don't like this, but I don't currently see a better way to do it. Maybe in the future we should add a cache?

Reply:
I don't particularly like it either but I don't think it's a big enough issue to warrant complicating things either. This works for now :shrug: